### ISSUE TITLE
Duplicate site search results

### ISSUE THREAD

**Issue by ryancreatescopy** [ethereum/ethereum-org-website#2764]
**Describe the bug**

Our search isn't behaving as intended.

We are seeing multiple results for seemingly the same page (try "dapps" for example) and when you click on one it seems to load for a microsecond before being replaced by the homepage.

When I check the URL of the page I end up on after clicking a search result it's https://ethereum.org/en#gatsby-focus-wrapper.

**Intended behaviour**
Search should only list each page once and should load the chosen page.

**Comment by minimalsm** [comment]
**tldr: this is two issues.**

- algolia search is treating ethereum.org/en/dapps as multiple pages
- links on this page are broken in the V2 translation only

-----

It seems to be the only content effected is on https://ethereum.org/en/dapps/. 

For some reason each category is being treated as its own individual page. Any content searched for on this page will produce 5 results.

![Screenshot 2021-03-31 at 21 09 05](https://user-images.githubusercontent.com/62268199/113204832-7196ea00-9265-11eb-9bba-218aa151e106.png)

The links produced by this search are:

https://ethereum.org/en/dapps/?category=finance#explore
https://ethereum.org/en/dapps/?category=collectibles#explore
https://ethereum.org/en/dapps/#explore
https://ethereum.org/en/dapps/?category=technology#explore
https://ethereum.org/en/dapps/?category=gaming#explore

The anchor (#explore) seems to be breaking the links causing a redirect to the root url + the anchor (e.g https://ethereum.org/en#explore in the case above.

### Other Languages

I search on tested the French and Spanish versions also. The behaviour is similar on the Spanish version (although only 4 results, instead of 5). Again, all links are broken.

![Screenshot 2021-03-31 at 21 28 38](https://user-images.githubusercontent.com/62268199/113207162-30eca000-9268-11eb-8594-1ce79697e5bc.png)

Similarly on the French version of the site there are duplicate results (4 again) however the links work as expected. 

![Screenshot 2021-03-31 at 21 31 19](https://user-images.githubusercontent.com/62268199/113207388-76a96880-9268-11eb-9355-0f574a04d520.png)
 
After testing this a bit further it seems that all translations have the issue of duplication but the broken links are exclusive to those where the dapps page is on the most modern (v2+ ?) translation (de, el, en, es, hi, hu, ro).  The other conditionally rendered pages are not affected in anyway.

Sadly I don't think I can take this any further as the as I don't have relevant algolia keys to test locally.

**Comment by IliadisVictor** [comment]
Hello there , I would like to be assigned on this issue I think I could work it out.

**Comment by samajammin** [comment]
Solved in #4220 - thanks @corwintines!
